pkg/watch/replay.go

```diff
@@ -121,94 +120,96 @@ func (r *replayTracker) replayWaitCh(gvk schema.GroupVersionKind) chan struct{}
 // replayEventsLoop processes requests to start and stop replaying events for
 // registrars that join an existing informer and need historical data.
 // Events for each registrar will be listed and replayed in an independent goroutine.
-func (wm *Manager) replayEventsLoop() error {
-	var wg sync.WaitGroup
-	defer wg.Wait()
+func (wm *Manager) replayEventsLoop(ctx context.Context) func() error {
+	return func() error {
+		var wg sync.WaitGroup
+		defer wg.Wait()
 
-	ctx, cancel := syncutil.ContextForChannel(wm.stopped)
-	defer cancel()
+		ctx, cancel := context.WithCancel(ctx)
```

Comment:
I think we could just use the parent context as-is?

Reply:
Looks like you're right - E2E tests pass.